usr/share/rear/lib/_framework-setup-and-functions.sh

```diff
@@ -1507,3 +1507,324 @@ function ProgressInfo () {
+    # see https://relax-and-recover.org/documentation/security-architecture
+    # SHARE_DIR and CONFIG_DIR and VAR_DIR are sub-directories of REAR_DIR_PREFIX:
+    TRUSTED_PATHS=( "$REAR_DIR_PREFIX/" '/usr/' '/etc/' '/lib/' )
+else
+    # When REAR_DIR_PREFIX is empty it is cases (A1) or (A2) i.e. no Git checkout
+    # or it is case (B2) i.e. in the ReaR recovery system when there was a Git checkout on the original system.
+    # Intentionally we do not test RECOVERY_MODE because this is not needed.
+    # Instead we test directly "the real thing" which is whether or not SHARE_DIR matches where its actual files are
+    # by checking an example (this script) i.e. if the actual path of /usr/share/rear/lib/_framework-setup-and-functions.sh
+    # is /usr/share/rear/lib/_framework-setup-and-functions.sh or /rear/prefix/usr/share/rear/lib/_framework-setup-and-functions.sh
+    # and in the latter case we specify '/rear/prefix/' first in trusted_paths
+    # cf. https://github.com/rear/rear/pull/3379#issuecomment-2601767515
+    # cf. the REAR_DIR_PREFIX setting code in sbin/rear
+    # TODO: Perhaps test_path="$SHARE_DIR/lib/_framework-setup-and-functions.sh" is better than hardcoded '/usr/share/rear'?
+    test_path="/usr/share/rear/lib/_framework-setup-and-functions.sh"
+    # Fall back to the normal cases (A1) or (A2) if 'readlink -e /usr/share/rear/lib/_framework-setup-and-functions.sh' fails:
+    actual_path="$( readlink -e "$test_path" )" || actual_path="$test_path"
+    if test "$actual_path" != "$test_path" ; then
+        # This is case (B1) i.e. in the ReaR recovery system when there was a Git checkout on the original system:
+        rear_dir_prefix=${actual_path%$test_path}
+        TRUSTED_PATHS=( "$rear_dir_prefix/" '/usr/' '/etc/' "$VAR_DIR/" '/lib/' )
+    else
+        # This is the normal cases (A1) or (A2) i.e. no Git checkout:
+        # In case of (A1) or (A2) SHARE_DIR is a sub-directory of '/usr/' and CONFIG_DIR is a sub-directory of '/etc/':
+        TRUSTED_PATHS=( '/usr/' '/etc/' "$VAR_DIR/" '/lib/' )
+    fi
+fi
+# 'readonly TRUSTED_PATHS' happens in sbin/rear
+# after the normal user configuration files (site.conf local.conf rescue.conf) were sourced
+# so that the user can specify what he needs, e.g. for whatever third-party (backup) software.
+
+# Check the actual file path (i.e. with symlinks resolved) is trusted
+# see https://github.com/rear/rear/issues/3259#issuecomment-2385745545
+function is_trusted_path () {
+    local file="$1"
+    local actual_path=""
+    local trusted_path=""
+    # Do not error out in 'readlink' when it is neither a regular file nor a link to a regular file
+    # but it is not trusted when it is neither a regular file nor a link to a regular file:
+    test -f "$file" || return 1
+    # Empty TRUSTED_PATHS means all regular file paths are blindly trusted,
+    # cf https://github.com/rear/rear/pull/3379#issuecomment-2633123866
+    test ${#TRUSTED_PATHS[@]} -eq 0 && return 0
```

Comment:
```suggestion
    # Empty TRUSTED_PATHS means all regular file paths are blindly trusted,
    # cf https://github.com/rear/rear/pull/3379#issuecomment-2633123866
    test ${#TRUSTED_PATHS[@]} -eq 0 && return 0
    # Do not error out in 'readlink' when it is neither a regular file nor a link to a regular file
    # but it is not trusted when it is neither a regular file nor a link to a regular file:
    test -f "$file" || return 1
```

Isn't it empty by default? So we can check for the common scenario first

Reply:
Thank you!
This was clearly the wrong ordering
regardless that by default neither TRUSTED_OWNERS
nor TRUSTED_PATH are empty - they are empty in
default.conf but this means their defaults are set in
_framework-setup-and-functions.sh which is sourced
in sbin/rear before the user configs gets sourced.